File: 1756-minimum-deletions-to-make-string-balanced/1756-minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int n = s.length();
        std::stack<char> charStack;
        int deleteCount = 0;

        // Iterate through each character in the string
        for (int i = 0; i < n; i++) {
            // If stack is not empty, top of stack is 'b', and current char is 'a'
            if (!charStack.empty() && charStack.top() == 'b' && s[i] == 'a') {
                charStack.pop();  // Remove 'b' from stack
                deleteCount++;    // Increment deletion count
            } else {
                charStack.push(s[i]);  // Push current character onto stack
            }
        }

        return deleteCount;
    }
};

File: 1756-minimum-deletions-to-make-string-balanced/1756-minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        n = len(s)
        a_count = sum(1 for ch in s if ch == 'a')

        b_count = 0
        min_deletions = n

        # Second pass: iterate through the string to compute minimum deletions
        for ch in s:
            if ch == 'a':
                a_count -= 1
            min_deletions = min(min_deletions, a_count + b_count)
            if ch == 'b':
                b_count += 1

        return min_deletions